### ISSUE TITLE
Code results in an ambiguous reference error

### ISSUE THREAD

**Issue by felus** [vapor/docs#315]
I followed the tutorial and at "Content" -> "Request" 
```swift
router.post("login") { req -> Future<HTTPStatus> in
    return req.content.decode(LoginRequest.self).map(to: HTTPStatus.self) { loginRequest in
        print(loginRequest.email) // user@vapor.codes
        print(loginRequest.password) // don't look!
        return .ok
    }
}
```

results first in an error that content.decode is using a throw, so it can easily be fixed encapsulating it in a try block. 
```swift
try {
    router.post("login") { req -> Future<HTTPStatus> in
      return req.content.decode(LoginRequest.self).map(to: HTTPStatus.self) { loginRequest in
          print(loginRequest.email) // user@vapor.codes
          print(loginRequest.password) // don't look!
          return .ok
      }
  }
}
```

furthermore the closure results in an ambiguous error, because the return value cannot be
determined. So put in a return value for the closure.
```swift
return req.content.decode(content: LoginRequest.self).map(to: HTTPStatus) { 
     LoginRequest -> HTTPStatus in
                print(loginRequest.email)
                print(loginRequest.password)
                return .ok
}
```

But this just descalades the ambiguous reference error and now the ".map" member says its ambiguous. I defined the return type of "decode", but the error persists.

TL;DR:
the official documentation has code which doesn't compile because the return type of HTTPRequestStatus cannot be predetermined by the compiler.

**Comment by felus** [comment]
so it builds now:

this is the fixed code.
`router.post("login") { req -> Future<HTTPStatus> in
        return try req.content.decode(LoginRequest.self).map(to: HTTPStatus.self) { loginRequest -> HTTPStatus in
            print(loginRequest.email)
            print(loginRequest.password)
            return .ok
        }
    }`

what i did: put the decode member into a try and determined the return value of the closure.

it builds now

**Comment by 0xTim** [comment]
Closing as this is old code but thank you for posting the solution @felus !
